charmcraft/commands/build.py

```diff
@@ -182,65 +183,64 @@ def run(self, bases_indices: Optional[List[int]] = None) -> List[str]:
         :returns: List of charm files created.
         """
         charms: List[str] = []
+        is_managed_mode = is_charmcraft_running_in_managed_mode()
 
-        if self.config.bases:
-            for bases_index, bases_config in enumerate(self.config.bases):
-                if bases_indices and bases_index not in bases_indices:
+        if not (self.charmdir / "charmcraft.yaml").exists() and not is_managed_mode:
+            notify_deprecation("dn02")
```

Comment:
This deprecation should come from `config.py`, which is the one handling `charmcraft.yaml`. I'm not sure why shouldn't be shown in managed mode...

Reply:
i put it here (for now at least) because of the statement in config that it can load for commands which do not require charmcraft.yaml.  I didn't think we would want DNs spewing for commands other than build/pack for now.

Reply:
Oh, you're right.